### PR TITLE
Remove app2 solicitor address textfield

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-705

### Change description ###

- Removes applicant 2 solicitor organisation text area field.
- Hidden field is used to store organisation contact information as it is used in generating draft application template. 
- CCD doesn't passes organisation information as part of organisation search component hence it retrieved through mid-event callback and set in an hidden field `applicant2OrgContactInformation`
- Tested on local docker.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/solicitor/client/organisation/OrganisationContactInformation.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class OrganisationContactInformation {
+    @CCD(
+        label = "Address Line 1"
+    )
+    @JsonProperty(value = "addressLine1")
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/client/organisation/OrganisationContactInformation.java:20]: these JsonProperty fields shouldn't be needed

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java [hunk 2]
```diff
@@ -41,4 +60,16 @@ public void addTo(final PageBuilder pageBuilder) {
             .optional(CaseData::getApplicant2HomeAddress, "applicant2SolicitorRepresented=\"No\"")
             .mandatory(CaseData::getApplicant2CorrespondenceAddress, "applicant2SolicitorRepresented=\"No\"");
     }
+
+    private AboutToStartOrSubmitResponse<CaseData, State> midEvent(
+        CaseDetails<CaseData, State> details,
+        CaseDetails<CaseData, State> detailsBefore
+    ) {
+        log.info("Mid-event callback triggered for Applicant2ServiceDetails");
+        return solicitorCreateApplicationService.setApplicant2SolOrganisationInfo(
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java:69]: I saw this in another place, I'm not sure how I feel about things in the Service layer returning `AboutToStartOrSubmitResponse` objects. It kind of feels like that stuff should be at the callback handler level but it's not super important.

File: src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorCreateApplicationService.java [hunk 3]
```diff
@@ -109,4 +111,29 @@ public CaseData aboutToSubmit(
             .data(caseData)
             .build();
     }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> setApplicant2SolOrganisationInfo(
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorCreateApplicationService.java:115]: In my opinion you should just move this method to the callback handler

**Review by linusnorton — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorCreateApplicationService.java (hunk 3 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorCreateApplicationService.java:115]: will move it as a separate PR

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java (hunk 2 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java:69]: will move it as a separate PR

**Pull request merged**